### DISCUSSION TITLE
Keep showing marker on screen if user clicks on a data point

### DISCUSSION THREAD

**Discussion by Praveen-Android** [patrykandpatrick/vico#868] [Questions]
### Question

Hello, I am new to this awesome library. In showcase app, I see that the marker disappears as soon as I move away from a data point. Is there a way to have the marker to keep showing on the screen when I click on a data point. 

I came across `persistentMarker` but that is more to persist a specific point at all times and not the point i select. `CartesianMarkerVisibilityListener` seems to be related to this but wanted to check if such a functionality is possible.

Thanks!

### Vico version(s)

2.0.0-alpha.28

### UI framework(s)

Jetpack Compose

**Comment by Tyler-Lopez** [comment]
I've found that I've needed to hack something around this: using only a persistent marker and passing in a transparent "normal marker" as the transparent normal marker moves I update the persistent marker position.

This should not be so hacky in the library imo, so I'd love to see a better way to do this.

**Reply by Praveen-Android** [reply]
Thanks @Tyler-Lopez. I tried it but wasn't able to get it to work probably because the api has changed w.r.t persistent markers. Would highly appreciate if you happen to have a sample code for me to try out. Thanks!

**Comment by Gowsky** [comment]
Hello, @Tyler-Lopez and @Praveen-Android.

> This should not be so hacky in the library imo, so I'd love to see a better way to do this.

It's hacky because it's a workaround — the library doesn't yet have the feature in question. We plan on adding it in the future, and once it's been introduced, this will be a matter of changing one argument. If you'd like to achieve this more robustly now, then rather than applying a workaround, you can subclass `CartesianChart` and make the required edits.

> I tried it but wasn't able to get it to work

Please show us what you tried.

**Comment by Tyler-Lopez** [comment]
Here is how you show a persistent marker at a point too, if you haven't seen this @Praveen-Android  https://github.com/patrykandpatrick/vico/blob/8dc63a34dbf8448da4b06a88d3cb3100a646a28b/sample/src/main/java/com/patrykandpatrick/vico/sample/showcase/charts/Chart1.kt#L88 

So, it's three steps: putting in a dummy transparent marker, creating a listener that updates an external mutable "show at", and then plugging in a separate persistent marker using that "show at" here.

**Comment by Gowsky** [comment]
@Praveen-Android, this seems to be a duplicate of #859. I've posted a comment explaining how to do this there — please see [here](https://github.com/patrykandpatrick/vico/discussions/859#discussioncomment-10860562). I'm closing this as a duplicate. If you have any further questions, please ask them under #859.
